src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -86,7 +86,9 @@
     ["extract" columns crit]
 
     [["extract" columns subquery]]
-    ["extract" columns ["and" subquery crit]]
+    (if (= "group_by" (first subquery))
```

Comment:
bit of a nitpick but can't we just do this with match like in the rest of this function?